File: app/src/main/java/com/example/olesya/boardgames/ui/view/SplashActivity.java
package com.example.olesya.boardgames.ui.view;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import com.example.olesya.boardgames.Entity.ImageHolder;
import com.example.olesya.boardgames.Utils;
import com.example.olesya.boardgames.database.AppDatabase;
import com.example.olesya.boardgames.database.ImageHolderDao;
import com.example.olesya.boardgames.ui.main.MainActivity;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.QueryDocumentSnapshot;

import java.util.ArrayList;

public class SplashActivity extends AppCompatActivity {

    private FirebaseFirestore db = FirebaseFirestore.getInstance();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ImageHolderDao dao = AppDatabase.Companion.getInstance(this).imagesDao();
        ArrayList<ImageHolder> holders = new ArrayList<>();
        db.collection(Utils.DB_PERSEPHONE)
                .get()
                .addOnCompleteListener(task -> {
                    if (task.isSuccessful()) {
                        for (QueryDocumentSnapshot document : task.getResult()) {
                            holders.add(new ImageHolder((String) document.getData().get("img")));
//                                Log.d(TAG, document.getId() + " => " + document.getData());
                        }
                    } else {
                        Toast.makeText(this, "trouble", Toast.LENGTH_SHORT).show();
                    }

                    insertData(dao, holders);
                    startActivity();
                });
    }

    private void insertData(ImageHolderDao dao, ArrayList<ImageHolder> holders) {
        new Thread(() -> dao.insertAll(holders)).start();
    }

    public void startActivity() {
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/example/olesya/boardgames/ui/view/SplashActivity.kt
package com.example.olesya.boardgames.ui.view

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.Toast
import com.example.olesya.boardgames.database.AppDatabase
import com.example.olesya.boardgames.database.ImageHolderDao
import com.example.olesya.boardgames.entity.ImageHolder
import com.example.olesya.boardgames.ui.main.MainActivity
import com.example.olesya.boardgames.ui.main.MenuViewModel.DB_PERSEPHONE
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.firestore.QueryDocumentSnapshot

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val firebaseDb = FirebaseFirestore.getInstance()
        val dao = AppDatabase.getInstance(this).imagesDao()
        val holders = mutableListOf<ImageHolder>()
        firebaseDb.collection(DB_PERSEPHONE)
                .get()
                .addOnCompleteListener { task ->
                    run {
                        if (task.isSuccessful) {
                            for (document: QueryDocumentSnapshot in task.result!!) {
                                holders.add(ImageHolder(document.data["img"] as String))
                            }
                        } else {
                            Toast.makeText(this, "trouble", Toast.LENGTH_SHORT).show();
                        }

                        insertData(dao, holders)
                        startActivity()
                    }
                }
    }

    private fun insertData(dao: ImageHolderDao, holders: MutableList<ImageHolder>) {
        dao.insertAll(holders)
    }

    private fun startActivity() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
    }
}